packages/gestalt/src/SideNavigation/GroupExpandIconButton.tsx

```diff
@@ -0,0 +1,76 @@
+import { Props as ItemContentProps } from './ItemContent';
+import Box from '../Box';
+import Icon from '../Icon';
+import IconButton, { Props as IconButtonProps } from '../IconButton';
+import useInteractiveStates from '../utils/useInteractiveStates';
+
+type Props = Pick<ItemContentProps, 'active'> & {
+  expanded: boolean;
+  isLink?: boolean;
+  expandIconButtonProps?: Pick<
+    IconButtonProps,
+    'accessibilityControls' | 'accessibilityExpanded' | 'onClick'
+  >;
+};
+
+export default function GroupExpandIconButton({
+  expanded,
+  active,
+  isLink: hasLink,
+  expandIconButtonProps,
+}: Props) {
+  const {
+    handleOnBlur,
+    handleOnFocus,
+    handleOnMouseEnter,
+    handleOnMouseLeave,
+    isHovered,
+    isFocused,
+  } = useInteractiveStates();
+  const activeIconButtonColor = isFocused || isHovered ? 'gray' : 'transparent';
+
+  if (!hasLink) {
+    return (
+      <Box
+        alignItems="center"
+        aria-hidden
+        display="flex"
+        height={24}
+        justifyContent="center"
+        marginEnd={-2}
+        rounding="circle"
+        tabIndex={-1}
+        width={24}
+      >
+        <Icon
+          accessibilityLabel=""
+          color={active ? 'inverse' : 'default'}
+          icon={expanded ? 'arrow-up' : 'arrow-down'}
+          size={12}
+        />
+      </Box>
+    );
+  }
+
+  return (
+    <IconButton
+      accessibilityControls={expandIconButtonProps?.accessibilityControls}
+      accessibilityExpanded={expandIconButtonProps?.accessibilityExpanded}
+      accessibilityLabel=""
```

Comment:
@hectoid i think we should label this. I want sure what to do here.

Reply:
Is the GroupExpandIconButton the caret that represents if the group is expanded or collapsed? 

If so, it does not need a label as the icon's focus target is wrapped up with the group button (although makes me wonder why this component uses an IconButton, and not just Icon), the state of the group is already communicated via `aria-expended`, see VO output:

<img width="894" alt="image" src="https://github.com/pinterest/gestalt/assets/82847291/e5c17845-7685-45fa-9931-e07e63bb84bd">

Reply:
This icons are more representational than anything else . the real button is the component itself. 

HOWEVER!!! If the whole think becomes a link while keeping the show/hide children, then this IconButton for opening then they have to 1. get the proper label (from provider) and 2. have keyboard navigation (they dont right now.)

![Brave Browser - SideNavigation - Gestalt 2024-06-21 at 10 49 28 AM](https://github.com/pinterest/gestalt/assets/10593890/94467a9e-f7d2-4723-b916-15344a4be230)

Reply:
@AlbertCarreras yes, you're correct. Actually this is the case where parent item is a link. I am just not sure what should be the default label